db/upgrade.php

```diff
@@ -341,5 +343,75 @@ function xmldb_cms_upgrade($oldversion) {
         upgrade_mod_savepoint(true, 2023112100, 'cms');
     }
 
+    if ($oldversion < 2023120101) {
+        // Add adhoc task to update contextid in customfield records for 'cmsfield' type.
+        manager::queue_adhoc_task(new update_customfield_context());
+        upgrade_mod_savepoint(true, 2023120101, 'cms');
+    }
+
+    if ($oldversion < 2023120102) {
+        $dbman = $DB->get_manager();
+        // Update valuetrust if exists for mod_cms (both 'cmsfield' and 'cmsuserlist').
+        if ($dbman->field_exists('customfield_data', 'valuetrust')) {
+            $sql = "SELECT mcd.id mcdid
+                      FROM {customfield_data} mcd
+                      JOIN {customfield_field} mcf ON mcf.id = mcd.fieldid
+                      JOIN {customfield_category} mcc ON mcc.id = mcf.categoryid
+                     WHERE mcc.component = 'mod_cms' AND mcd.valuetrust = 0";
+            $records = $DB->get_records_sql($sql);
+            $mcdids = array_keys($records);
+            foreach (array_chunk($mcdids, 100) as $ids) {
```

Comment:
maybe we can make chunks bigger?  There are 74 085 records that need to be updated

Reply:
Increase chunk from 100 to 1000.